### DISCUSSION TITLE
SSH.NET 2024.1 encryption features.

### DISCUSSION THREAD

**Discussion by rud0x7b6** [sshnet/SSH.NET#1437] [Q&A] [answered]
Hi,

I have a list of encryption features and I would want to be sure that SSH.NET 2024.1 supports them. I made some investigation and it seems to me that SSH.NET supports all of them, but could you help me a bit and confirm it? I mean using of SSH.NET in Net Framework 4.8 application, not in Net Core project.

The list:
1. RSA 2048 bits
2. AES 256 
3. HMAC-SHA256
4. RSA encryption (2048 bits) of a SHA-256 digest)
5. The certificate must be compliant with the X.509v3 standard.
6. ECDH-256 or alternatively Diffie-Hellman (2048 bits).

**Comment by Rob-Hague** [comment, marked as answer]
5 is not relevant unless you are talking about OpenSSH certificate authentication, the rest is OK

**Reply by rud0x7b6** [reply]
Thank you.
